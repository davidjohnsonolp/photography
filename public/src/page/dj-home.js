import buildShadowRoot from '../lib/build-shadow-root.js';

class DJHome extends HTMLElement {
  constructor() {
    super();
    const html = `
      <style>
        :host {
          
        }
      
        dj-footer {
          position: absolute;
          bottom: 0;
          width: 100%;
        }
      </style>
      <dj-header transparent="true" animate="true">
      </dj-header>

      <main></main>

      <dj-footer transparent="true" animate="true">
      </dj-footer>
    `;
    buildShadowRoot(html,this);
  }
}

customElements.define('dj-home', DJHome);
export default DJHome;
