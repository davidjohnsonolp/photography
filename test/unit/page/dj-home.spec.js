import '../../../public/src/page/dj-home.js';

let homePage;

describe('DJHome Home Page', () => {
  beforeEach(() => {
    homePage = window.document.createElement('dj-home'); 
    document.body.appendChild(homePage);
  });

  afterEach(() => {
    document.body.getElementsByTagName('dj-home')[0].remove();
  });

  it('renders correctly', () => {
    expect({ html: homePage.shadowRoot.innerHTML }).toMatchSnapshot();
  });
});