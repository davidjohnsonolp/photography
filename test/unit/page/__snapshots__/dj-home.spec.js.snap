// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DJHome Home Page renders correctly 1`] = `
Object {
  "html": "
      <style>
        :host {
          
        }
      
        dj-footer {
          position: absolute;
          bottom: 0;
          width: 100%;
        }
      </style>
      <dj-header transparent=\\"true\\" animate=\\"true\\">
      </dj-header>

      <main></main>

      <dj-footer transparent=\\"true\\" animate=\\"true\\">
      </dj-footer>
    ",
}
`;
